static_files/assets/js/script-manager.js

```diff
@@ -756,197 +758,188 @@ const eventLocationTypes = function ($locations) {
+    }).on('typeahead:render', (_event, suggestions, finished) => {
+      if (finished === false) {
+        return;
+      }
+
+      if (suggestions.length < 1) {
+        $addNewTagBtn.show();
       } else {
-        return false;
+        $addNewTagBtn.hide();
       }
-    };
-
-    // Custom function for removing selected tag list items.
-    // This function DOES update self.selectedTagsArray.
-    autocomplete.removeTag = function (listItem) {
-      const self = this;
-      self.selectedTagsList
-        .find(listItem)
-        .remove();
-
-      const item = listItem.attr('data-tag-name');
-      self.selectedTagsArray.splice($.inArray(item, self.selectedTagsArray), 1);
-    };
-
-    // Custom function that returns a 'clean' string value, after running through
-    // a regular expression to only allow whitelisted characters below
-    autocomplete.getCleanItemVal = function (item) {
-      return $.trim(item.replace(/([^a-zA-Z0-9\s-!$#%&+|:?])/g, ''));
-    };
-
-    autocomplete.setupForm = function () {
-      const self = this;
-      if (self.dataField.is(':visible')) {
-        // Get existing tags from self.dataField and push them to self.selectedTagsArray.
-        // Populate the selectedTagsList with existing tags.
-        let existingTaglistVal = '';
-        if (self.dataField.val()) {
-          existingTaglistVal = self.dataField.val();
-        } else if (self.dataField.attr('value')) {
-          existingTaglistVal = self.dataField.attr('value');
-        }
-        if (existingTaglistVal !== '') {
-          // Create array from existingTaglistVal. $.grep removes empty results.
-          const tagArray = $.grep(existingTaglistVal.replace(/(&quot;?)|"/g, '').split(','), (val) => {
-            return val !== '';
-          });
-          self.selectedTagsArray = self.selectedTagsArray.concat(tagArray);
-
-          if (self.selectedTagsList.children().length < 1) {
-            $.each(tagArray, (key, val) => {
-              self.createTag(self.getCleanItemVal(val));
-            });
-          }
-        }
+    });
+  };
 
-        // Enable autocomplete field. Hide data field.
-        self.dataField.hide();
-        self.autocompleteField
-          .insertAfter(self.dataField)
-          .show();
-        $(`label[for="${self.dataField.attr('id')}"]`).attr('for', self.autocompleteField.attr('id'));
+  /**
+   * Handles adding a tag to the selectedTags
+   * array as well as the unordered list of tags.
+   * @param {any} suggestion The tag objects being added
+   * @returns {void}
+   */
+  const addTagItem = (suggestion) => {
+    if (selectedTags.indexOf(suggestion.text) > -1) {
+      return;
+    }
 
-        // Insert "Add Tag" btn onto page
-        self.addBtn
-          .insertAfter(self.autocompleteField)
-          .hide();
+    selectedTags.push(suggestion.text);
+    const $removeLink =
+      $('<a href="#" class="text-inverse" alt="Remove this tag" title="Remove this tag">&times</a>')
```

Comment:
We can use fontawesome icons here and on the add button in order to match the styles of the instance remove/add buttons. Added mr-1 as well so it's a bit separated from the text. Let's also add back in the `selected-remove` class and remove the text-inverse so that the already-defined hover styles for this are included (https://github.com/UCF/unify-events/blob/rc-v2.2.0/static_files/assets/scss/components/_event-create-update.scss#L5-L14).
```suggestion
      $('<a href="#" class="selected-remove" alt="Remove this tag" title="Remove this tag"><span class="fa fa-times mr-1" aria-hidden="true"></span></a>')
```

Reply:
Updated.